app/experimenter/docs/adrs/0004-monitoring-data-import.md

```diff
@@ -0,0 +1,67 @@
+# Monitoring Data Import
+
+* Status: proposed
+* Deciders: Anna Scholtz, Jared Kerim, Tim Smith
+* Date: 2020-11-12
+
+
+## Context and Problem Statement
+
+Currently, Jetstream is responsible for daily processing experiment data for analysis after enrollment has completed. We're also interested in processing experiment data for monitoring purposes. Monitoring data is a little different from analysis data in that it is live (updated every few minutes) and needs to be available immediately when an experiment starts collecting data. Jetstream is only responsible for generating analysis data and we need an alternative approach to access monitoring data from Experimenter.
+
+Currently, [Grafana](https://grafana.telemetry.mozilla.org) is used to display monitoring data. It works by querying live tables from BigQuery on-demand and does not have any caching. We would like to eventually replace Grafana capabilities with custom visualizations inside of Experimenter. The goal is to have more flexibility than Grafana gives for how to display the data as well as less overhead for users when context switching from one visualization platform to another.
+
+## Decision Drivers
+
+* Data needs to be displayed "live", updating within minutes
+* Cost needs to stay low
+* Have flexibility with how the data is visualized
+* Data loading speed
+
+
+## Considered Options
+
+1. Add ETL in [biguqery_etl](https://github.com/mozilla/bigquery-etl) to process the live data periodically and output results to a gcs bucket that can be accessed by Experimenter. This is how Jetstream data is currently being accessed.
+2. Add a background celery task that fetches data on-demand from bigquery and stores it in Redis. Redis cache can be used for the next fetch of the same data, but it gets invalidated periodically.
+3. Embed existing Grafana graphs (e.g. [graphs](https://grafana.telemetry.mozilla.org/d/XspgvdxZz/experiment-enrollment?orgId=1&var-experiment_id=bug-1656561-rapid-testing-demo-nightly-80&from=1596153600000&to=1598832000000))
+
```

Comment:
I'd like to know if there's a way to do it by running rolling accumulators for enrollment/unenrollment events at the pubsub level that could compute continuous population levels for every live experiment that would give us an immediate sense of every live experiments enrollment without doing it at the bq level, either on demand or scheduled etl?

Reply:
When I was looking at this earlier, reading the event pings from pubsub would be prohibitively expensive – this might change if/when we move to pubsub lite, but we'd need to do some analysis on costs. The other mark against this is that it's a special case that would increase maintenance burden vs using existing, established tooling.

Reply:
Ah that's too bad, then if it has to be done at the bq level then @emtwo 's right we should just do it on demand with a celery task 👍